src/EnergyPlus/SurfaceGeometry.cc

```diff
@@ -1455,11 +1456,9 @@ namespace SurfaceGeometry {
             ShowSevereError(state, format("{}Reordered # of Surfaces ({}) not = Total # of Surfaces ({})", RoutineName, MovedSurfs, TotSurfaces));
             SurfError = true;
             for (int Loop = 1; Loop <= TotSurfaces; ++Loop) {
-                if (state.dataSurfaceGeometry->SurfaceTmp(Loop).Class != SurfaceClass_Moved) {
-                    if (state.dataSurfaceGeometry->SurfaceTmp(Loop).Class > 100) {
-                        ShowSevereError(state, RoutineName + "Error in Surface= \"" + state.dataSurfaceGeometry->SurfaceTmp(Loop).Name + "\" Class=" +
-                                        cSurfaceClass(state.dataSurfaceGeometry->SurfaceTmp(Loop).Class - 100) + " indicated Zone=\"" + state.dataSurfaceGeometry->SurfaceTmp(Loop).ZoneName + "\"");
-                    }
+                if (!SurfaceTmpClassMoved(Loop) && state.dataSurfaceGeometry->SurfaceTmp(Loop).Class == SurfaceClass::INVALID) {
+                    ShowSevereError(state, RoutineName + "Error in Surface= \"" + state.dataSurfaceGeometry->SurfaceTmp(Loop).Name + "\" Class=" +
+                                        cSurfaceClass(state.dataSurfaceGeometry->SurfaceTmp(Loop).Class) + " indicated Zone=\"" + state.dataSurfaceGeometry->SurfaceTmp(Loop).ZoneName + "\"");
```

Comment:
Hmmm, INVALID is getting used as an array index here in the error message. Sorry. I guess you should just go back to the bool array you had before for invalid.

Reply:
That's just a lookup to the string name of INVALID, which you can just hardcode as INVALID. That's not a good enough reason to go back to an arrays of bools.

Reply:
Uhhh.. The `SurfaceTmp(Loop).Class` it wants to print here was the original surface class. (SurfaceTmp(Loop).Class - 100)
```
if (state.dataSurfaceGeometry->SurfaceTmp(Loop).Class > 100) {
ShowSevereError(state, RoutineName + "Error in Surface= \"" + state.dataSurfaceGeometry->SurfaceTmp(Loop).Name + "\" Class=" +
cSurfaceClass(state.dataSurfaceGeometry->SurfaceTmp(Loop).Class - 100) + " indicated Zone=\"" + state.dataSurfaceGeometry->SurfaceTmp(Loop).ZoneName + "\"");
 }
```
Can I just skip printing the Class in the error message? Surface Name and Zone Name should be enough to identify it?

Reply:
I should also say that adding 100 to SurfaceClass to indicate an error is exemplary programming. Exemplary of what not to do that is. If you want to retrieve the original class of the surface, just make a boolean array to indicate that there was an error. And don't do arithmetic on SurfaceClass (not saying that you, Xuan, did this to begin with).

Reply:
@xuanluo113 Sure, just drop the class from the error message - good idea.